docs/sources/mimir/operators-guide/architecture/query-sharding/index.md

```diff
@@ -139,6 +139,17 @@ parallel by the query-frontend, multiplying the previously set value of
 `-querier.max-query-parallelism` by
 `-query-frontend.query-sharding-total-shards`.
 
+## Cardinality-estimation for query sharding (experimental)
+Since a high number of parallel sharded queries increases load on queriers and their dependencies, it is desirable to shard queries only as much as necessary.
+Queries that return more series (i.e., are of high cardinality) need to fetch more data and should therefore be split into a larger batch of shards, while queries that return few or no series should be executed with fewer or no shards at all.
+When determining the number of shards to use for a given query, the sharding logic can optionally take into account the cardinality observed during previous executions of the same query for similar time ranges.
+
+To enable this experimental feature, set `-query-frontend.query-sharding-max-series-per-shard` to a value representing roughly how many series each shard should fetch, and enable result caching (the result cache is used to store cardinality estimates).
+The value set for this flag is one of several parameters used by the sharding logic to determine the appropriate number of shards for a query.
```

Comment:
```suggestion
The value that you set for this flag is one of several parameters that sharding logic uses to determine the appropriate number of shards for a given query.
```

Reply:
Shouldn't it be 
```suggestion
The value that you set for this flag is one of several parameters that the sharding logic uses to determine the appropriate number of shards for a query.
```
?

Also am I correct to assume that to your generalized feedback here is to avoid the passive voice as much as possible?

Reply:
Yes.